src/dumpsxiso/main.cpp

```diff
@@ -314,18 +333,23 @@ static EntryAttributes EstablishXMLAttributeDefaults(tinyxml2::XMLElement* defau
 	// First establish "defaults" - that is, the most commonly occurring attributes
 	auto findMaxElement = [](const auto& map)
 	{
-		return std::max_element(map.begin(), map.end(), [](const auto& left, const auto& right) { return left.second < right.second; })->first;
+		if (map.size() > 0) {
+			return std::max_element(map.begin(), map.end(), [](const auto& left, const auto& right) { return left.second < right.second; })->first;
+		}
+		return 0;
 	};
 
 	EntryAttributes defaultAttributes;
 	defaultAttributes.GMTOffs = static_cast<signed char>(findMaxElement(attributeCounters.GMTOffs));
+	defaultAttributes.FFLAGS = static_cast<unsigned char>(findMaxElement(attributeCounters.FFLAGS));
 	defaultAttributes.XAAttrib = static_cast<unsigned char>(findMaxElement(attributeCounters.XAAttrib));
 	defaultAttributes.XAPerm = static_cast<unsigned short>(findMaxElement(attributeCounters.XAPerm));
 	defaultAttributes.GID = static_cast<unsigned short>(findMaxElement(attributeCounters.GID));
 	defaultAttributes.UID = static_cast<unsigned short>(findMaxElement(attributeCounters.UID));
 
 	// Write them out to the XML
 	defaultAttributesElement->SetAttribute(xml::attrib::GMT_OFFSET, defaultAttributes.GMTOffs);
+	defaultAttributesElement->SetAttribute(xml::attrib::FILE_FLAGS, (defaultAttributes.FFLAGS < 2) ? defaultAttributes.FFLAGS : defaultAttributes.FFLAGS - 2);
```

Comment:
This `- 2` looks magic, what does it signify?

Reply:
This is for the hidden attribute flag, but I currently changed it to
if (defaultAttributes.HFLAG & 0x01) {
	defaultAttributesElement->SetAttribute(xml::attrib::HIDDEN_FLAG, 1);
}